polar-core/src/partial/simplify.rs

```diff
@@ -30,6 +30,10 @@ pub struct Simplifier;
 impl Folder for Simplifier {
     fn fold_term(&mut self, t: Term) -> Term {
         match t.value() {
+            Value::Partial(Constraints { operations, .. }) if operations.is_empty() => {
```

Comment:
A partial with no constraints is replaced with a boolean true.  I'm not sure I understand this simplification. Could you explain further?

Couldn't it be any value?

Reply:
Constraints are conjunctive truth conditions. If there are no constraints, the truth conditions are vacuous, and we have a zero-argument `and`, which evaluates to `true`.

You could substitute any trivially true expression, such as `1 = 1`, but only `true` can't be simplified any further.

Reply:
What about this knowledge base:

```
f(x);
```

query: `f(partial(x))`

Output `x = true`

query: `f(x)` (x is a variable)
output: `x = Variable('_x_2')`

These seem inconsistent.

Reply:
A possibly more consistent answer would be a partial with an empty constraint set. But we've decided that such partials, and ones that unify against ground terms, should be maximally simplified.